### ISSUE TITLE
Utilizing multiple function/tools calls?

### ISSUE THREAD

**Issue by devjn** [BrainBlend-AI/atomic-agents#31]
Can we use function calls instead of JSON mode? Particularly would be nice to get example with Vertex AI Gemini as they don't support Union types.

Currently I can see that when specified Mode.VERTEXAI_TOOLS for instructor,  and then for BaseAgentConfig(output_schema=AgentOutputSchema)
I can see from logs that 1 tool is defined like this:
'tools': [function_declarations {
 name: "AgentOutputSchema
 ...
 }]

How could we define more tools and instruct correctly?

**Comment by KennyVaneetvelde** [comment]
Hello, unless I misunderstand your question, I believe this question is more suited for https://github.com/instructor-ai/instructor 

Maybe though it's not possible with Vertex AI in tool mode, either way I'm sure the folks at Instructor are more knowledgeable on this specific subject!

**Comment by devjn** [comment]
Yeah it is seems to be more question for instructor. However, currently there is [parallel-function-calling](https://python.useinstructor.com/concepts/parallel/#understanding-parallel-function-calling) that takes `Iterable` so to utilize it we would need `output_schema` to allow it. Something like this: `BaseAgentConfig(output_schema=Iterable[WeatherSchema | GoogleSearchSchema])`

**Comment by KennyVaneetvelde** [comment]
@devjn Have you tried this? Anything that works for instructor should work within atomic agents, or is there something specific that's not working or holding you back here?

**Comment by devjn** [comment]
@KennyVaneetvelde The issue currently is that `BaseAgent` expects BaseAgentConfig for output_schema and if I use Iterable I get: `ValidationError: 1  validation error for BaseAgentConfig output_schema Input should be a subclass of BaseModel`
Or should I not use `BaseAgent`?

**Comment by KennyVaneetvelde** [comment]
Right, The config is just for configuration, but the point still stands as right now it only supports a class that extends from Pydantic's `BaseModel`  but I get what you mean, you need it to accept an iterable...

I will try to look into that!

**Comment by KennyVaneetvelde** [comment]
In retrospect, I'd suggest creating a model and having the iterable be a property of that model, that way you can later add stuff more easily, so this is to be considered the more idiomatic approach
